Mathlib/RingTheory/IntegralRestrict.lean

```diff
@@ -119,3 +124,337 @@ lemma prod_galRestrict_eq_norm [IsGalois K L] [IsIntegrallyClosed A] (x : B) :
   rw [← IsScalarTower.algebraMap_apply, IsScalarTower.algebraMap_eq A K L]
   simp only [map_prod, algebraMap_galRestrict_apply, IsIntegralClosure.algebraMap_mk',
     Algebra.norm_eq_prod_automorphisms, AlgHom.coe_coe, RingHom.coe_comp, Function.comp_apply]
+
+end galois
+
+attribute [local instance] FractionRing.liftAlgebra FractionRing.isScalarTower_liftAlgebra
+
+noncomputable
+instance (priority := 900) [IsDomain A] [IsDomain B] [IsIntegrallyClosed B]
+    [Module.Finite A B] [NoZeroSMulDivisors A B] : Fintype (B ≃ₐ[A] B) :=
+  haveI : IsIntegralClosure B A (FractionRing B) :=
+    IsIntegralClosure.of_isIntegrallyClosed _ _ _ (Algebra.IsIntegral.of_finite A B)
+  haveI : IsLocalization (Algebra.algebraMapSubmonoid B A⁰) (FractionRing B) :=
+    IsIntegralClosure.isLocalization _ (FractionRing A) _ _
+      (isAlgebraic_of_isFractionRing _ _ (Algebra.IsIntegral.of_finite A B))
+  haveI : FiniteDimensional (FractionRing A) (FractionRing B) :=
+    Module.Finite_of_isLocalization A B _ _ A⁰
+  Fintype.ofEquiv _ (galRestrict A (FractionRing A) (FractionRing B) B).toEquiv
+
+variable {Aₘ Bₘ} [CommRing Aₘ] [CommRing Bₘ] [Algebra Aₘ Bₘ] [Algebra A Aₘ] [Algebra B Bₘ]
+variable [Algebra A Bₘ] [IsScalarTower A Aₘ Bₘ] [IsScalarTower A B Bₘ]
+variable (M : Submonoid A) [IsLocalization M Aₘ]
+variable [IsLocalization (Algebra.algebraMapSubmonoid B M) Bₘ]
+
+section trace
+
+/-- The restriction of the trace on `L/K` restricted onto `B/A` in an AKLB setup.
```

Comment:
Is "AKLB setup" standard terminology (in mathlib)?
It's the first time I see it.

Reply:
```
In this file, we assume `A` is an integrally closed domain; `K` is the fraction ring of `A`;
`L` is a finite (separable) extension of `K`; `B` is the integral closure of `A` in `L`.
We call this the AKLB setup.
```
is in the module docstring.

Reply:
I see. That makes sense. But I'm not sure that we should assume this kind of info in docstrings, since they show up outside the context of the file.
Maybe add
```suggestion
/-- The restriction of the trace on `L/K` restricted onto `B/A` in an AKLB setup (see module docstring).
```
for now...

Reply:
I think this terminology also appears in some literature, for example Ash's A Course In Algebraic Number Theory.
Maybe I can expand the docstring into a library note in a future PR and reference it here. 
Have we implemented library note in lean4 yet?

Reply:
I kept the docstring as is for now. This is an auxiliary result that probably won't be used outside this file.